andela/Ah-backend-xmen
# #162725264 users should favorite articles

#### What does this PR do?
This pr adds functionality to enable authenticated users to favorite unfavorite articles. 

#### Description of the task to be implemented.
Articles that tickle our fancy are worth favoriting and those deemed distasteful are worth unfavoriting.

#### How should this be manually tested?
- Fetch this branch to your local machine by running `git fetch origin ft-favorite-articles-162725264`  on your terminal
- Check into this branch by running `git checkout ft-favorite-articles-162725264` 
- Run `python manage.py migrate` to introduce a new field
- Run your server by running `python manage.py runserver`
- Favorite an article by making a `POST` request on any article `/api/articles/article-slug/favorite`
![screenshot 2019-01-25 at 20 01 08](https://user-images.githubusercontent.com/18280598/51760686-0c229e00-20dc-11e9-9034-e96d6548fe31.png)


- View favorite  articles by making a `GET` request on any article `/api/articles/favorites`
<img width="1135" alt="screenshot 2019-01-28 at 11 07 24" src="https://user-images.githubusercontent.com/18280598/51822369-1d8ec480-22ed-11e9-8ebf-860f7374fe36.png">

- Unfavorite an article by making a `DELETE` request on any article `/api/articles/article-slug/favorite`
![screenshot 2019-01-25 at 20 04 09](https://user-images.githubusercontent.com/18280598/51760860-98cd5c00-20dc-11e9-844c-4b83b405f8ea.png)



#### Related pivotal tracker stories.
[#162725264](https://www.pivotaltracker.com/story/show/162725264)

#### Postman Collections
[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/9720cdb1ee29cf146477)
